### PR TITLE
[CULT-136] - feature/merge coreos@v2.9.0

### PR DESCRIPTION
**Pull request by ghost**
* Merges `coreos@v2.9.0`.
* Reverts `coreos` -> `liquidlabs-co` rename.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 PROJ=dex
-ORG_PATH=github.com/liquidlabs-co
+ORG_PATH=github.com/coreos
```
Comment (fridiculous) [inline Makefile:2]: I think this file, and others, need this ORG_PATH to liquidlabs-co for the go code to compile locally

Comment (fridiculous) [reply Makefile:2]: It's the main reason i moved everything to liquidlabs-co

**Comment by fridiculous** [comment]
Because this PR is so core to authentication of our hosting cluster, i'd like to spin up a test cluster via kops to actually test this PR before merging.

File: Makefile (hunk 1 above)
Comment (ghost) [reply Makefile:2]: I was able to get it to compile.

```bash
export GOPATH=$HOME/go
go get -u github.com/Masterminds/glide
go get -u github.com/sgotti/glide-vc
make revendor
make docker-image
```

**Comment by ghost** [comment]
Totally agree. It will need testing.

**Comment by jramos** [comment]
@fridiculous -- ready for some testing. I "reused" this repo, but I don't think we need to build dex ourselves, only the example app and the new dockerfile.

```
$ cd gde-dex/
$ docker build .
Sending build context to Docker daemon  157.2kB
Step 1/3 : FROM quay.io/coreos/dex:v2.9.0
 ---> 6fec0d7fcf4e
Step 2/3 : COPY . .
 ---> 8cfca4eafa8d
Step 3/3 : EXPOSE 5556
 ---> Running in 4007fa51cbe9
Removing intermediate container 4007fa51cbe9
 ---> 43d7521899cc
Successfully built 43d7521899cc

$ docker run 43d7521899cc serve config.yaml
time="2018-02-02T21:55:24Z" level=info msg="config issuer: https://dex.gde.gigsternetwork.com:5556"
time="2018-02-02T21:55:24Z" level=info msg="config storage: sqlite3"
time="2018-02-02T21:55:24Z" level=info msg="config static client: example-app"
time="2018-02-02T21:55:24Z" level=info msg="config connector: mock"
time="2018-02-02T21:55:24Z" level=info msg="config connector: local passwords enabled"
time="2018-02-02T21:55:24Z" level=info msg="keys expired, rotating"
time="2018-02-02T21:55:24Z" level=info msg="keys rotated, next rotation: 2018-02-03 03:55:24.433966995 +0000 UTC"
time="2018-02-02T21:55:24Z" level=info msg="listening (http) on 0.0.0.0:5556"
```

And then visit:

http://127.0.0.1:5556/auth?client_id=example-app&redirect_uri=http%3A%2F%2F127.0.0.1%3A5555%2Fcallback&response_type=code&scope=openid+profile+email+offline_access&state=I+wish+to+wash+my+irish+wristwatch

cc: @chasestarr

**Comment by ghost** [comment]
(We'll need to update `config.yaml` to contain whatever is already in k8s. I just copied the entire `examples/k8s` folder from `coreos` and modified enough to get working locally.)

**Comment by jramos** [comment]
`example-app` can be built via the following:

```
make build
make docker-image-example-app
```

**Comment by ghost** [comment]
Nevermind. We need a custom image due to the way github usernames are handled. Both dex and the example app should be working now. I've pushed tags to both repos for `v2.9.0`.

**Comment by chasestarr** [comment]
Can we find some time today to QA this together?

**Comment by ghost** [comment]
To run dex v2.9.0 using the ECR images:

```bash
export ECR_BASE=569325332953.dkr.ecr.us-east-1.amazonaws.com

docker pull $ECR_BASE/dex:v2.9.0
docker pull $ECR_BASE/dex-signin:v2.9.0
docker run $ECR_BASE/dex:v2.9.0 serve config.yaml
docker run $ECR_BASE/dex-signin:v2.9.0 --issuer <URL for dex>
```

To build the images locally:

```bash
export GOPATH=$HOME/go

make docker-image
make docker-image-example-app
```

**Pull request merged**